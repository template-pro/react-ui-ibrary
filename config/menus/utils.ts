const utilsMenu = [
  {
    title: 'General',
    children: ['src/with-default-props'],
  },
  {
    title: 'React Hooks',
    children: ['src/use-props-value'],
  },
  // {
  //   title: "Date & Time",
  //   children: [],
  // },
  // {
  //   title: "String",
  //   children: [],
  // },
]

export default utilsMenu
