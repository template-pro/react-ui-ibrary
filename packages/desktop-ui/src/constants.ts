export const defaultPrefixCls = 'temp-pc'
