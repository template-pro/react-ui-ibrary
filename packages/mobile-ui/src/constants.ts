export const defaultPrefixCls = "jgb-m"