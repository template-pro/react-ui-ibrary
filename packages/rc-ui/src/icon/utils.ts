import { useEffect } from 'react'
import { insertCss } from 'insert-css'

export const svgBaseProps = {
  // width: '1em',
  // height: '1em',
  'fill': 'currentColor',
  'aria-hidden': 'true',
  'focusable': 'false',
}

export const iconStyles = `
.template-pro-icon {
  display: inline-block;
  width: 24px;
  color: inherit;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.template-pro-icon > * {
  line-height: 1;
}

.template-pro-icon svg {
  display: inline-block;
}

.template-pro-icon::before {
  display: none;
}

.template-pro-icon .template-pro-icon {
  display: block;
}

.template-pro-icon[tabindex] {
  cursor: pointer;
}

.template-pro-icon-spin::before,
.template-pro-icon-spin {
  display: inline-block;
  -webkit-animation: loadingCircle 1s infinite linear;
  animation: loadingCircle 1s infinite linear;
}

@-webkit-keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
`

let cssInjectedFlag = false

export const useInsertStyles = (styleStr = iconStyles) => {
  useEffect(() => {
    if (!cssInjectedFlag) {
      insertCss(styleStr, {
        prepend: true,
      })
      cssInjectedFlag = true
    }
  }, [])
}
