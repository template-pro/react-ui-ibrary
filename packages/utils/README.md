# @template-pro/utils

统一工具库，通用正则，React Hooks 等工具。

## 安装

```bash
npm install @template-pro/utils
```
