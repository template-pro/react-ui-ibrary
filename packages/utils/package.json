{
  "name": "@template-pro/utils",
  "version": "0.0.0",
  "description": "统一工具库，通用正则，react hooks 等工具",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./es/index.d.ts",
  "files": [
    "lib",
    "es"
  ],
  "scripts": {
    "build": "gulp",
    "release":"bumpp --tag @template-pro/utils@%s --all --no-verify && pnpm publish"
  },
  "keywords": [
    "utils"
  ],
  "author": "Wuxh<wxh1220@gmail.com>",
  "license": "MIT",
  "publishConfig": {
    "registry": "http://nas.wxhboy.cn:98/"
  }
}
