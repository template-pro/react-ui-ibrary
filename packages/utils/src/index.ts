export { default as usePropsValue } from './use-props-value'
export { default as mergeProps } from './with-default-props'

// RegExp
export * as anyRule from './any-rule'
