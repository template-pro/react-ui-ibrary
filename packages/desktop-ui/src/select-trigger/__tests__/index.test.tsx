import React from 'react';
import SelectTrigger from '..'
import { fireEvent, render } from '../../../../../tests/utils';
import { defaultPrefixCls } from '../../constants';

describe('SelectTrigger', () => {
  it('正确 render', () => {
    const { container } = render(<SelectTrigger />)
    expect(container).toMatchSnapshot()
  })

  it('点击时应该触发点击事情', () => {
    const onClick = jest.fn()
    const { container } = render(<SelectTrigger onClick={onClick} />)
    fireEvent.click(container.firstChild as HTMLElement)
    expect(onClick).toHaveBeenCalled()
  })

  it('支持数组和字符串格式 value', () => {
    const stringValue = 'hello world'

    const arrayValue = [
      { value: 'hello' },
      { value: 'world' },
    ]

    const { container, rerender } = render(<SelectTrigger value={stringValue} />)

    const getValueNode = (body: any) => body.querySelector(`.${defaultPrefixCls}-select-trigger-value`)
    expect(getValueNode(container)).toMatchSnapshot()

    rerender(<SelectTrigger value={arrayValue} />)
    expect(getValueNode(container)).toMatchSnapshot()
  })
})