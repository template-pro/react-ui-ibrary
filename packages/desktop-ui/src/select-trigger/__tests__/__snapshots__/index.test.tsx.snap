// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTrigger 支持数组和字符串格式 value 1`] = `
<p
  class="template-pro__pc-select-trigger-value"
>
  hello world
</p>
`;

exports[`SelectTrigger 支持数组和字符串格式 value 2`] = `
<p
  class="template-pro__pc-select-trigger-value"
>
  hello
  、
  world
</p>
`;

exports[`SelectTrigger 正确 render 1`] = `
<div>
  <div
    class="template-pro__pc-select-trigger template-pro__pc-select-trigger-size__lg"
    tabindex="0"
  >
    <p
      class="template-pro__pc-select-trigger-placeholder"
    />
    <span
      class="template-pro__pc-select-trigger-suffix"
    >
      <span
        class="template-pro__pc-select-trigger-select-text"
      >
        选择
      </span>
    </span>
  </div>
</div>
`;
